django_mongodb_backend/query_conversion/expression_converters.py

```diff
@@ -0,0 +1,129 @@
+"""Expression To Match Converters"""
+
+
+class _BaseExpressionConverter:
+    """
+    Base class for optimizers that handle specific operations in MQL queries.
+    This class can be extended to implement optimizations for other operations.
+    """
+
+    @classmethod
+    def convert(cls, expr):
+        raise NotImplementedError("Subclasses should implement this method.")
+
+    @classmethod
+    def is_simple_value(cls, value):
+        """
+        Check if the value is a simple type (not a dict).
+        """
+        if isinstance(value, str) and value.startswith("$"):
+            return False
+        if isinstance(value, list | tuple | set):
+            return all(cls.is_simple_value(v) for v in value)
+        # TODO: Expand functionality to support `$getField` conversion
+        return not isinstance(value, dict) or value is None
+
+    @classmethod
+    def is_convertable_field_name(cls, field_name):
+        """Validate a field_name is one that can be represented in $match"""
+        # This needs work and re-evaluation
+        return (
+            isinstance(field_name, str)
+            and field_name.startswith("$")
+            and not field_name[:1].isalnum()
+        )
+
+
+class _EqExpressionConverter(_BaseExpressionConverter):
+    """Convert $eq operation to a $match compatible query."""
```

Comment:
For this classes it would be helpful to include small examples of what the transformation before+after looks like. Is that possible?